atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/SweepQueueUtils.java

```diff
@@ -76,6 +81,20 @@ public static WriteInfo toWriteInfo(TableReference tableRef, Map.Entry<Cell, byt
         return WriteInfo.of(WriteReference.of(tableRef, cell, isTombstone), timestamp);
     }
 
+    public static void validateNumberOfCellsWritten(Collection<List<WriteInfo>> writes) {
+        writes.forEach(writesForShard -> {
+            int cellsWritten = writesForShard.size();
+            if (cellsWritten > MAX_CELLS_PER_SHARD) {
```

Comment:
this is precise, of course, but also weirdly probabilistic (fails based on hash function). Fancy changing it to just sum the total writes? Seems reasonable to say 'Atlas supports up to 6.4MM writes in a transaction'.

Reply:
otherwise you get this based on the murmur hash of your keys which is a bit weird

Reply:
I generally agree with this, but are we sure we want to be this heavy handed? As it is, we always have an escape hatch where we can increase the sharding using the runtime config

Reply:
@gmaretic any resolution on this?

Reply:
I'm going ahead and merge as is, since we couldn't get a conclusive ack that 6.4M is high enough. We can revisit at some later point if we really want to